### PR TITLE
[FLINK-2834] Global round-robin for temporary directories

### PR DESCRIPTION
**Pull request by greghogan**
Multiple TaskManager filesystems can be used by configuring multiple temporary directories.
This patch changes the process of spilling files from a per-operator round-robin to a global
round-robin such that each directory is written to in turn across all operators, reducing
unbalanced I/O due to bunching.

### CODE REVIEW

**Comment by StephanEwen** [comment]
Looks good to me, +1 to merge this!

File: flink-runtime/src/main/java/org/apache/flink/runtime/io/disk/iomanager/FileIOChannel.java [hunk 1]
```diff
@@ -154,20 +155,25 @@ private static String randomString(Random random) {
...
 
 		private final File[] paths;
-		
+
 		private final String namePrefix;
 
-		private int counter;
+		private int localCounter;
+
+		private static AtomicInteger globalCounter = new AtomicInteger();
```
Comment (StephanEwen) [inline flink-runtime/src/main/java/org/apache/flink/runtime/io/disk/iomanager/FileIOChannel.java]: Minor (optional) comment: In the (undocumented) code style that Flink follows implicitly, the static member fields are the first in the class, before the instance member fields.

Comment (greghogan) [inline flink-runtime/src/main/java/org/apache/flink/runtime/io/disk/iomanager/FileIOChannel.java]: Fixed.

**Comment by fhueske** [comment]
Will merge this PR

**Pull request closed without merging**